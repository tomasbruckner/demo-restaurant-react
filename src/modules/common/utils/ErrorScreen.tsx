import React, { PureComponent } from 'react';
import PageLayout from '../layouts/PageLayout';
import { I18N_CORE } from 'core/i18n/i18n';
import { WithTranslation, withTranslation } from 'react-i18next';

export class ErrorScreen extends PureComponent<{ title: string } & WithTranslation> {
  render() {
    const { t } = this.props;

    return <PageLayout title={t('page.error.title')} />;
  }
}

export default withTranslation(I18N_CORE)(ErrorScreen);
