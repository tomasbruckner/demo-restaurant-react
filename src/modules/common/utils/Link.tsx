import { Link as RouterLink } from 'react-router-dom';

export default RouterLink;
