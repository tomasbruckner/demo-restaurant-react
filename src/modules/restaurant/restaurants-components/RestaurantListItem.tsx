import React, { PureComponent } from 'react';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import { Typography, withStyles } from '@material-ui/core';
import { I18N_CORE } from 'core/i18n/i18n';
import { WithTranslation, withTranslation } from 'react-i18next';
import RatingWrapper from '../common-components/RatingWrapper';
import { RouteComponentProps, withRouter } from 'react-router';
import { getRestaurantDetail } from 'core/router/urls';
import { RestaurantResponse } from 'services/restaurantService';
import { API_ENDPOINT } from 'config/config';
import Avatar from '@material-ui/core/Avatar';
import RestaurantMenuIcon from '@material-ui/icons/RestaurantMenu';

const StyledAvatar = withStyles({
  root: {
    height: '100%',
    width: '100%',
    cursor: 'pointer',
  },
})(Avatar);

const StyledMenuIcon = withStyles({
  root: {
    height: '100%',
    width: '100%',
  },
})(RestaurantMenuIcon);

export type Props = RestaurantResponse;

export class RestaurantListItem extends PureComponent<
  Props & WithTranslation & RouteComponentProps<{}>
> {
  redirect = () => {
    const { restaurantId, history } = this.props;

    history.push(getRestaurantDetail(restaurantId));
  };

  render() {
    const { name, description, numberOfReviews, averageRating, imageLink } = this.props;

    return (
      <Card
        style={{ margin: '16px', maxWidth: '40%' }}
        onClick={this.redirect}
        data-cy="restaurant-list-item"
      >
        <CardActionArea style={{ display: 'flex' }}>
          <StyledAvatar
            src={`${API_ENDPOINT}${imageLink}`}
            variant="square"
            style={{ height: '100%', maxHeight: '250px', maxWidth: '250px' }}
          >
            <StyledMenuIcon />
          </StyledAvatar>
          <CardContent>
            <Typography gutterBottom variant="h4" component="h2">
              {name}
            </Typography>
            <RatingWrapper
              averageRating={averageRating}
              numberOfReviews={numberOfReviews}
              canEdit={false}
            />
            <Typography variant="body2" component="p">
              {description}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    );
  }
}

export default withTranslation(I18N_CORE)(withRouter(RestaurantListItem));
