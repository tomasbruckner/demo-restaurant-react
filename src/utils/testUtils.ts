export const fakeTranslate = (key: string) => key;

export const fakeLoad = () => Promise.resolve();

export const fakeLogger = {
  info() {},
  error() {},
};

// eslint-disable-next-line no-unused-vars
export const fakeFunction = (): any => {};

export const fakeHistory = {
  push: () => {},
};

export const fakeTranslationProps = {
  t: fakeTranslate,
  i18n: {} as any,
  tReady: true,
};

export const fakeReview = {
  reviewId: 1,
  reviewRatingId: 2,
  restaurantId: 3,
  title: 'title',
  text: 'text',
  dateOfVisit: '',
  hasOwnerResponse: true,
  created: '',
  owner: {
    userId: 4,
    firstName: 'firstName',
    lastName: 'lastName',
    imageLink: 'imageLink',
  },
  ownerResponse: {
    reviewReplyId: 5,
    reviewId: 6,
    creatorId: 7,
    created: '',
    updated: '',
    text: 'text',
  },
  restaurantName: 'restaurantName',
};

export const fakeUser = {
  userId: 1,
  roleId: 2,
  firstName: 'firstName',
  lastName: 'lastName',
  email: 'email@email',
  created: '',
  imageLink: 'string',
};

export const fakeRestaurant = {
  restaurantId: 1,
  ownerId: 2,
  name: 'name',
  description: 'description',
  websiteUrl: 'websiteUrl',
  contactPhone: 'contactPhone',
  address: 'address',
  numberOfReviews: 3,
  averageRating: 4,
  hasCurrentUserReviewed: true,
  imageLink: 'imageLink',
};

export const fakeRouterProps = {
  location: {} as any,
  match: {} as any,
  history: { push: () => {} } as any,
};
