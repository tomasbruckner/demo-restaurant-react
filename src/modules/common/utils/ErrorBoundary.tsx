import React, { PureComponent } from 'react';
import { WithTranslation, withTranslation } from 'react-i18next';
import Logger from 'utils/logger';
import { I18N_CORE } from 'core/i18n/i18n';
import ErrorScreen from './ErrorScreen';

export class ErrorBoundaryComp extends PureComponent<
  { renderProp?: () => React.ReactNode } & WithTranslation
> {
  state = {
    error: null,
  };

  static getDerivedStateFromError(error: Error) {
    return { error };
  }

  componentDidCatch(error: unknown, errorInfo: unknown) {
    Logger.error(error, errorInfo);
  }

  render() {
    const { children, renderProp, t } = this.props;
    const { error } = this.state;

    if (!error) {
      return children;
    }

    if (renderProp) {
      return renderProp();
    }

    return <ErrorScreen title={t('CommonComponents.ErrorBoundary.error')} />;
  }
}

export default withTranslation(I18N_CORE)(ErrorBoundaryComp);
