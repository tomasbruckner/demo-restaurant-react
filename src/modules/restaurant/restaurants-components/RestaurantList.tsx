import React, { PureComponent } from 'react';
import RestaurantListItem, { Props as RestaurantProps } from './RestaurantListItem';
import { PagedDto } from 'types/commonTypes';

type Props = {
  restaurants: PagedDto<RestaurantProps>;
};

export class RestaurantList extends PureComponent<Props> {
  render() {
    const { restaurants } = this.props;

    return restaurants.data.map(props => (
      <RestaurantListItem key={props.restaurantId} {...props} />
    ));
  }
}

export default RestaurantList;
