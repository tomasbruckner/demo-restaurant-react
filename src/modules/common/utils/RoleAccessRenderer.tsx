import React, { PureComponent, ReactNode } from 'react';
import { connect, ConnectedProps } from 'react-redux';
import { RootState } from 'core/redux/store';
import { getUserFromState } from 'core/redux/ducks/UserDuck';
import { Role } from 'utils/constants';

const connector = connect((state: RootState) => {
  return {
    user: getUserFromState(state),
  };
});

type Props = {
  onlyAdmin?: boolean;
  adminOrOwner?: boolean;
  renderNotAllowed?: ReactNode;
  children: React.ReactNode;
};

class RoleAccessRenderer extends PureComponent<Props & ConnectedProps<typeof connector>> {
  checkPermissions = () => {
    const { user, onlyAdmin, adminOrOwner } = this.props;

    if (!user) {
      return false;
    }

    if (onlyAdmin) {
      return user.roleId === Role.Admin;
    }

    if (adminOrOwner) {
      return user.roleId === Role.Admin || user.roleId === Role.Owner;
    }

    return false;
  };

  render() {
    const { children, renderNotAllowed = null } = this.props;

    if (this.checkPermissions()) {
      return children;
    }

    return renderNotAllowed;
  }
}

export default connector(RoleAccessRenderer);
