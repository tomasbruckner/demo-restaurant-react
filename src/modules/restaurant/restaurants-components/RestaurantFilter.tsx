import React, { PureComponent } from 'react';
import { I18N_CORE } from 'core/i18n/i18n';
import { WithTranslation, withTranslation } from 'react-i18next';
import Typography from 'modules/common/components/controls/Typography';
import Rating from 'modules/common/components/controls/Rating';
import { connect, ConnectedProps } from 'react-redux';
import { restaurantFilterAsync } from 'core/redux/ducks/RestaurantDuck';
import { DEFAULT_PAGE_SIZE } from 'utils/constants';
import styled from 'styled-components/macro';

const connector = connect(null, {
  dispatchFilterRestaurants: restaurantFilterAsync,
});

const StyledDiv = styled.div`
  margin: 32px;
`;

export class RestaurantFilter extends PureComponent<
  WithTranslation & ConnectedProps<typeof connector>
> {
  onChange = (event: React.ChangeEvent<{}>, value: number | null) => {
    const { dispatchFilterRestaurants } = this.props;

    dispatchFilterRestaurants({
      take: DEFAULT_PAGE_SIZE,
      sort: {
        predicate: 'averageRating',
        reverse: true,
      },
      filter: {
        averageRating: value,
      },
    });
  };

  render() {
    const { t } = this.props;

    return (
      <StyledDiv>
        <Typography>{t('component.restaurant-filter.rating-filter')} </Typography>
        <Rating precision={0.5} onChange={this.onChange} name="rating" />
      </StyledDiv>
    );
  }
}

export default withTranslation(I18N_CORE)(connector(RestaurantFilter));
