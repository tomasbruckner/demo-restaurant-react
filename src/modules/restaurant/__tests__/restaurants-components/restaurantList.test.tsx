import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { RestaurantList } from 'modules/restaurant/restaurants-components/RestaurantList';
import { fakeTranslationProps } from 'utils/testUtils';

describe('RestaurantList', () => {
  it('basic example', () => {
    const tree = renderer
      .create(
        <RestaurantList
          restaurants={{ size: 0, status: '', data: [] }}
          {...fakeTranslationProps}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
