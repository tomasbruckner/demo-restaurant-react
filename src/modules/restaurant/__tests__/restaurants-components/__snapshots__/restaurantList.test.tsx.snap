// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantList basic example 1`] = `null`;
